x/slashing/keeper.go

```diff
@@ -111,23 +110,25 @@ func (k Keeper) handleValidatorSignature(ctx sdk.Context, addr crypto.Address, p
 	// This counter just tracks the sum of the bit array
 	// That way we avoid needing to read/write the whole array each time
 	previous := k.getValidatorSigningBitArray(ctx, consAddr, index)
-	if previous == signed {
+	missed := !signed
+	if previous == missed {
 		// Array value at this index has not changed, no need to update counter
-	} else if previous && !signed {
-		// Array value has changed from signed to unsigned, decrement counter
-		k.setValidatorSigningBitArray(ctx, consAddr, index, false)
-		signInfo.SignedBlocksCounter--
-	} else if !previous && signed {
-		// Array value has changed from unsigned to signed, increment counter
+	} else if !previous && missed {
+		// Array value has changed from not missed to missed, increment counter
 		k.setValidatorSigningBitArray(ctx, consAddr, index, true)
```

Comment:
`missed` is true which means the validator doesn't sign the current validator. So I thinks we should set sign bit array to false:
```
k.setValidatorSigningBitArray(ctx, consAddr, index, false)
```

Reply:
I changed the semantics, so this array is now tracking whether a validator *missed* the block instead of whether they *signed* the block. Maybe the name of the function should be changed too for clarify.

Reply:
So updated.